rust/sdk/src/app_manifest.rs

```diff
@@ -1,29 +1,81 @@
 use crate::{app_id, AppId};
+use core::convert::TryFrom;
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Deserialize, Serialize, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
 #[serde(rename_all = "camelCase")]
-/// App manifest used for requesting a bearer token via the API. For more information see the
-/// [docs](https://developer.actyx.com/docs/how-to/app-auth/authenticate-with-app-manifest).
-pub struct AppManifest {
-    /// App id in lower case and in reverse domain name notation
+struct AppManifestIo {
     pub app_id: AppId,
-    /// Display name of the app
     pub display_name: String,
-    /// Version string of the app
     pub version: String,
-    /// Developer certificate's signature
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub signature: Option<String>,
 }
 
+impl From<AppManifest> for AppManifestIo {
+    fn from(value: AppManifest) -> Self {
+        value.0
+    }
+}
+
+impl TryFrom<AppManifestIo> for AppManifest {
+    type Error = String;
+    fn try_from(value: AppManifestIo) -> Result<Self, Self::Error> {
+        if value.signature.is_none() && !value.app_id.starts_with("com.example.") {
+            Err(format!(
+                "Trial app id needs to start with 'com.example.'. Got '{}'.",
+                value.app_id
+            ))
+        } else {
+            Ok(Self(value))
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(into = "AppManifestIo", try_from = "AppManifestIo")]
+pub struct AppManifest(AppManifestIo);
+
 impl AppManifest {
-    pub fn new(app_id: AppId, display_name: String, version: String, signature: Option<String>) -> Self {
-        AppManifest {
+    pub fn signed(app_id: AppId, display_name: String, version: String, signature: String) -> Self {
+        Self(AppManifestIo {
             app_id,
             display_name,
             version,
-            signature,
+            signature: Some(signature),
+        })
+    }
+
+    pub fn trial(app_id: AppId, display_name: String, version: String) -> anyhow::Result<Self> {
+        if !app_id.starts_with("com.example.") {
+            anyhow::bail!("Trial app id needs to start with 'com.example.'. Got '{}'.", app_id);
         }
```

Comment:
Instead of requiring this from the user, we should just take the last part of the ID and always prepend  `com.example.` to make it work